### PR TITLE
Add section for ICU dependent APIs

### PR DESCRIPTION
**Pull request by tarekgh**
## Summary

Adding more info to the `.NET globalization and ICU` doc.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/extensions/globalization-icu.md](https://github.com/dotnet/docs/blob/132d3a574de3bdc2daeefa063115952e5b92502e/docs/core/extensions/globalization-icu.md) | [.NET globalization and ICU](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu?branch=pr-en-us-36377) |

### CODE REVIEW

**Comment by tarekgh** [comment]
@gewarren could you please help review this change too? Thanks!

File: docs/core/extensions/globalization-icu.md [hunk 1]
```diff
@@ -57,6 +57,21 @@ You can run code analysis rules [CA1307: Specify StringComparison for clarity](.
...
+#### TimeZoneInfo.FindSystemTimeZoneById
+
+ICU provides the flexibility to create TimeZoneInfo instances using [IANA](https://www.iana.org/time-zones) time zone IDs, even when the application is running on Windows. Similarly, it enables the creation of TimeZoneInfo instances with Windows time zone IDs, even when running on non-Windows platforms. However, it's important to note that this functionality will not be available when using [NLS mode](https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#use-nls-instead-of-icu) or [Globalization Invariant Mode](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md).
+
+#### ICU dependent APIs
+
+.NET introduced APIs that are dependent on ICU. These APIs can succeed only when using ICU. Here are some examples:
+
+- [TryConvertIanaIdToWindowsId](https://learn.microsoft.com//dotnet/api/system.timezoneinfo.tryconvertianaidtowindowsid)
+- [TryConvertWindowsIdToIanaId](https://learn.microsoft.com//dotnet/api/system.timezoneinfo.tryconvertwindowsidtoianaid)
```
Comment (gewarren) [inline docs/core/extensions/globalization-icu.md:69]:
```suggestion
- <xref:System.TimeZoneInfo.TryConvertWindowsIdToIanaId%2A>
```

File: docs/core/extensions/globalization-icu.md [hunk 2]
```diff
@@ -57,6 +57,21 @@ You can run code analysis rules [CA1307: Specify StringComparison for clarity](.
...
+
+If the application is running on Windows 10 May 2019 Update or any newer versions, the mentioned APIs will consistently succeed. However, on older versions of Windows, these APIs will consistently fail. In such cases, users have the option to enable the [App-local ICU](https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#app-local-icu) feature to ensure the success of these APIs. On non-Windows platforms, these APIs will always succeed regardless of the version.
```
Comment (gewarren) [inline docs/core/extensions/globalization-icu.md]:
```suggestion
On Windows 10 May 2019 Update or any later versions, the mentioned APIs will consistently succeed. However, on older versions of Windows, these APIs will consistently fail. In such cases, you can enable the [app-local ICU](#app-local-icu) feature to ensure the success of these APIs. On non-Windows platforms, these APIs will always succeed regardless of the version.
```

File: docs/core/extensions/globalization-icu.md [hunk 3]
```diff
@@ -57,6 +57,21 @@ You can run code analysis rules [CA1307: Specify StringComparison for clarity](.
...
+
+In addition, it is crucial for apps to ensure that they are not running in [Globalization Invariant mode](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md) or [NLS mode](https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#use-nls-instead-of-icu) to guarantee the success of these APIs.
```
Comment (gewarren) [inline docs/core/extensions/globalization-icu.md]:
```suggestion
In addition, it's crucial for apps to ensure that they're not running in [globalization invariant mode](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md) or [NLS mode](#use-nls-instead-of-icu) to guarantee the success of these APIs.
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**